sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ExecutingSqlQueries_Net46.cs

```diff
@@ -181,5 +181,17 @@ public void NonCompliant_SqlCeCommands(SqlCeConnection connection, SqlCeTransact
             var command = new SqlCeCommand($"SELECT * FROM mytable WHERE mycol={param}"); // Noncompliant {{Make sure that executing SQL queries is safe here.}}
             command.CommandText = string.Format("INSERT INTO Users (name) VALUES (\"{0}\")", param); // Noncompliant
         }
+
+        public void Bar(SqlConnection connection, string param)
+        {
+            SqlCommand command;
+            string sensitiveQuery = string.Format("INSERT INTO Users (name) VALUES (\"{0}\")", param);
+            command = new SqlCommand(sensitiveQuery); // Noncompliant
+
+            command.CommandText = sensitiveQuery; // Noncompliant
+
+            SqlDataAdapter adapter;
+            adapter = new SqlDataAdapter(sensitiveQuery, connection); // Noncompliant
+        }
```

Comment:
could you also add tests with the FNs for class fields and properties and also 
```
        public void Bar(SqlConnection connection, string param)
        {
            SqlCommand command;
            string sensitiveQuery = string.Format("INSERT INTO Users (name) VALUES (\"{0}\")", param);
			string x = sensitiveQuery;

            command.CommandText = x; // FN
        }
```

and also
```
        public void Bar(SqlConnection connection, string param)
        {
            SqlCommand command;
            string sensitiveQuery;
			sensitiveQuery = string.Format("INSERT INTO Users (name) VALUES (\"{0}\")", param);

            command.CommandText = x; // FN
        }
```

Reply:
The first case is actually supported since `x` is an identifier and in this case we will recursively check also it's declaration (see the switch from `IsStringMethodInvocation`) but you're right and this needs to be documented also by tests.